Fix publish-docs

Fix `publish-docs` in the workflow. `ImportError: cannot import name 'environmentfilter' from 'jinja2' (/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/jinja2/__init__.py)` appears in https://github.com/E3SM-Project/zstash/runs/5795869500?check_suite_focus=true. This issue was introduced with the merging of #205.

@tomvothecoder Looks like the fixes in #207 didn't work: https://github.com/E3SM-Project/zstash/runs/5821359209?check_suite_focus=true fails with:
```
Run pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 jinja2<3.1
  pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 jinja2<3.1
  shell: /usr/bin/bash -l {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.12/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.12/x64/lib
/home/runner/work/_temp/757fff76-faa8-4d7a-9725-b4631e956582.sh: line 1: 3.1: No such file or directory
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build_workflow.yml
+++ b/.github/workflows/build_workflow.yml
@@ -81,7 +81,7 @@ jobs:
 
       - name: Install Dependencies
         run: |
-          pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 jinja2<3.1
+          pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 "jinja2<3.1"
 
       - name: Build Sphinx Docs
         run: |

--- a/.github/workflows/release_workflow.yml
+++ b/.github/workflows/release_workflow.yml
@@ -37,7 +37,7 @@ jobs:
       # Low risk trade-off with mismatching dependencies in `conda/zstash_dev.yml`
       - name: Install Dependencies
         run: |
-          pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 jinja2<3.1
+          pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4 docutils==0.16 "jinja2<3.1"
 
       - name: Build Sphinx Docs
         run: |

